pymodbus/transport/transport.py

```diff
@@ -468,9 +468,8 @@ async def do_relisten(self) -> None:
             Log.debug("Wait 1s before reopening listener.")
             await asyncio.sleep(1)
             await self.listen()
-        except asyncio.CancelledError:
-            pass
-        self.reconnect_task = None
+        finally:
+            self.reconnect_task = None
```

Comment:
This is wrong, because it breaks to code, self.reconnect_delay_current is never reset and the listeners are not popped.